### PR TITLE
Rename `ActionParam` to `ItemAction`

### PR DESCRIPTION
**Pull request by AngheloAlf**
~~Yet another Player preparation PR~~

The `ItemParam` name is not final, the purpose of this PR is to keep this discussion going so this doesn't get buried in the depths of Discord.

The main motivation to rename this away from "Action" is because it is confusing considering we usually use the "Action" word as in "ActionFuncs", meaning actual actions the actor performs, but in the case of Player the current ActionParams does not cover all the possible actions he can perform, it isn't even related to his actual `actionFunc` callback.

So, since all those "params" are related to item usage we have decided to name it after items (except for the zora fins outlier (which kinda is a repurposed boomerang, so it should be an item... whatever)).

I'm not completely convinced by `ItemParam` (mainly because `IP` reminds me of "Internet Protocol"), but I don't have any better idea. Any other suggestion is welcomed.

### CODE REVIEW

File: src/code/z_player_lib.c [hunk 1]
```diff
@@ -727,7 +727,7 @@ s32 Player_IsBurningStickInRange(PlayState* play, Vec3f* pos, f32 xzRange, f32 y
     Vec3f diff;
     s32 pad;
 
-    if ((this->itemActionParam == PLAYER_AP_STICK) && (this->unk_B28 != 0)) {
+    if ((this->itemActionParam == PLAYER_IP_STICK) && (this->unk_B28 != 0)) {
```
Comment (mzxrules) [inline src/code/z_player_lib.c:730]: ItemActionParam?

**Comment by EllipticEllipsis** [comment]
Or just plain `ItemAction`, not convinced `param` is actually doing anything here... it's not really a parameter.

**Comment by hensldm** [comment]
`ItemAction` seems good to me.

**Comment by Dragorn421** [comment]
May I bring your attention to `ACTORCAT_ITEMACTION` already using "item action" ?

**Comment by mzxrules** [comment]
Having done a quick look at where `ACTORCAT_ITEMACTION` is being used, it's pretty clear that `ACTORCAT_ITEMACTION` should be renamed.

**Review by hensldm — APPROVED** [review]

**Review by Dragorn421 — COMMENTED** [review]
+1 on "item action"

File: src/code/z_player_lib.c [hunk 2]
```diff
@@ -602,110 +602,110 @@ s32 func_80123810(PlayState* play) {
...
+    PLAYER_MODELGROUP_DEFAULT,        // PLAYER_IA_PICTO_BOX
+    PLAYER_MODELGROUP_INSTRUMENT,     // PLAYER_IA_OCARINA
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_FISH
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_SPRING_WATER
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_HOT_SPRING_WATER
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_ZORA_EGG
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_DEKU_PRINCESS
+    PLAYER_MODELGROUP_BOTTLE,         // PLAYER_IA_BOTTLE_GOLD_DUST
     PLAYER_MODELGROUP_BOTTLE,         // PLAYER_AP_BOTTLE_1C
```
Comment (EllipticEllipsis) [inline src/code/z_player_lib.c:650]: There's a few missed in here.

File: src/code/z_player_lib.c [hunk 3]
```diff
@@ -1384,68 +1384,68 @@ s32 func_80124110(Player* player, PlayerActionParam actionParam) {
...
 
     return false;
 }
 
-s32 Player_ActionToBottle(Player* player, PlayerActionParam actionParam) {
-    s32 bottle = actionParam - PLAYER_AP_BOTTLE;
+s32 Player_ActionToBottle(Player* player, PlayerItemAction itemAction) {
+    s32 bottle = itemAction - PLAYER_IA_BOTTLE;
 
     // Relies on bottle-related action params to be contiguous
```
Comment (EllipticEllipsis) [inline src/code/z_player_lib.c:1415]: Here too.

File: src/code/z_player_lib.c [hunk 4]
```diff
@@ -1384,68 +1384,68 @@ s32 func_80124110(Player* player, PlayerActionParam actionParam) {
...
-    return Player_ActionToBottle(Player, Player->itemActionParam);
+    return Player_ActionToBottle(Player, Player->heldItemAction);
 }
 
-s32 Player_ActionToExplosive(Player* player, PlayerActionParam actionParam) {
-    s32 explosive = actionParam - PLAYER_AP_BOMB;
+s32 Player_ActionToExplosive(Player* player, PlayerItemAction itemAction) {
+    s32 explosive = itemAction - PLAYER_IA_BOMB;
 
     // Relies on explosive-related action params to be contiguous
```
Comment (EllipticEllipsis) [inline src/code/z_player_lib.c:1430]: And here, you'd better do a global search.

File: tools/namefixer.py [hunk 5]
```diff
@@ -783,21 +785,98 @@
...
+    "PLAYER_AP_FISHING_POLE": "PLAYER_IA_FISHING_ROD",
+    "PLAYER_AP_MINUS1": "PLAYER_IA_MINUS1",
+    "PLAYER_AP_NONE": "PLAYER_IA_NONE",
+    "PLAYER_AP_LAST_USED": "PLAYER_IA_LAST_USED",
+    "PLAYER_AP_FISHING_ROD": "PLAYER_IA_FISHING_ROD",
+    "PLAYER_AP_SWORD_KOKIRI": "PLAYER_IA_SWORD_KOKIRI",
+    "PLAYER_AP_SWORD_RAZOR": "PLAYER_IA_SWORD_RAZOR",
+    "PLAYER_AP_SWORD_GILDED": "PLAYER_IA_SWORD_GILDED",
+    "PLAYER_AP_SWORD_GREAT_FAIRY": "PLAYER_IA_SWORD_GREAT_FAIRY",
+    "PLAYER_AP_STICK": "PLAYER_IA_STICK",
```
Comment (EllipticEllipsis) [inline tools/namefixer.py:811]: Isn't there a simpleReplace list so you can just do `PLAYER_AP_ : PLAYER_IA_`?

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Agreed with renaming Player Action to Player Item Action.

**Pull request merged**